engine/rpcserver.go

```diff
@@ -1600,55 +1600,65 @@ func (s *RPCServer) GetAuditEvent(_ context.Context, r *gctrpc.GetAuditEventRequ
 
 // GetHistoricCandles returns historical candles for a given exchange
 func (s *RPCServer) GetHistoricCandles(_ context.Context, req *gctrpc.GetHistoricCandlesRequest) (*gctrpc.GetHistoricCandlesResponse, error) {
-	if req.Exchange == "" {
+	if !req.UseDb && req.Exchange == "" {
 		return nil, errors.New(errExchangeNameUnset)
 	}
 
 	if req.Pair.String() == "" {
 		return nil, errors.New(errCurrencyPairUnset)
 	}
 
-	exchange := GetExchangeByName(req.Exchange)
-	if exchange == nil {
-		return nil, errors.New("Exchange " + req.Exchange + " not found")
-	}
-
 	var candles kline.Item
 	var err error
-	if req.ExRequest {
-		candles, err = exchange.GetHistoricCandlesExtended(currency.Pair{
-			Delimiter: req.Pair.Delimiter,
-			Base:      currency.NewCode(req.Pair.Base),
-			Quote:     currency.NewCode(req.Pair.Quote),
-		},
+	resp := gctrpc.GetHistoricCandlesResponse{
+		Interval: kline.Interval(req.TimeInterval).Short(),
+		Start:    req.Start,
+		End:      req.End,
+	}
+
+	if req.UseDb {
+		candles, err = kline.LoadFromDatabase(req.Exchange,
+			currency.Pair{
+				Delimiter: req.Pair.Delimiter,
+				Base:      currency.NewCode(req.Pair.Base),
+				Quote:     currency.NewCode(req.Pair.Quote),
+			},
 			asset.Item(strings.ToLower(req.AssetType)),
+			kline.Interval(req.TimeInterval),
```

Comment:
Something needs to be investigated here. Using the following:
```
.\gctcli.exe gethistoriccandlesextended -e=huobi --pair=BTC-USDT --asset=spot -interval=60 -sync=true
```
Save data to the database.
Using:
```
.\gctcli.exe gethistoriccandlesextended -e=huobi --pair=BTC-USDT --asset=spot -interval=60 -db=true
```

Cannot retrieve data

Reply:
Going to need more information including logs with debug set to true
I running the exact same commands I get data returned from the database there is also test coverage for LoadFromDatabase() that confirms data is returned correctly

Reply:
Looks potentially time related based on the generated query
```
[DEBUG] | DATABASE | 19/08/2020 12:55:23 | SQL: SELECT * FROM "candle" WHERE (base = ?) AND (quote = ?) AND (interval = ?) AND (asset = ?) AND (timestamp between ? and ?) AND (exchange_id = ?);
[DEBUG] | DATABASE | 19/08/2020 12:55:23 | SQL: [BTC USDT 60 spot 2020-07-19 22:55:23 +1000 AEST 2020-08-19 22:55:23 +1000 AEST 364fa4f1-407e-482e-b74d-860d95e84630]
```

![image](https://user-images.githubusercontent.com/9261323/90586757-51a53f80-e21b-11ea-946c-e2b86a0e11a5.png)
![image](https://user-images.githubusercontent.com/9261323/90586773-579b2080-e21b-11ea-93c3-6ecbb429b475.png)

Reply:
Pushed a fix up I believe (from my own testing) addresses this 
was an issue with timestamps and sqlite as you mentioned

Reply:
Good stuff! Can confirm it works 🎉